packages/components/src/custom-select/index.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * External dependencies
+ */
+import Downshift, { useSelect } from 'downshift';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { Button, Dashicon } from '../';
+
+const itemToString = ( item ) => item.name;
+// This is needed so that in Windows, where
+// the menu does not necessarily open on
+// key up/down, you can still switch between
+// options with the menu closed.
+const stateReducer = (
+	{ selectedItem },
+	{ type, changes, props: { items } }
+) => {
+	switch ( type ) {
+		case Downshift.stateChangeTypes.ToggleButtonKeyDownArrowDown:
```

Comment:
it's actually `useSelect.stateChangeTypes` here and below.

Reply:
Thank you!